pennylane/transforms/qcut/montecarlo.py

```diff
@@ -517,6 +517,11 @@ def _wrapper(*args, **kwargs):
 
         execute_kwargs["cache"] = False
 
+        if "mode" in execute_kwargs:
+            mode = execute_kwargs.pop("mode")
+            if not qml.active_return():
+                execute_kwargs["grad_on_execution"] = "forward" if mode != "best" else "best"
```

Comment:
Do we only use `grad_on_execution` when the new return types are disabled?

Reply:
The grad on exeution is only given with the new return system so here I convert  mode to the grad on execution.